cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -269,7 +275,9 @@ public synchronized void updateProperties(Id.Namespace namespaceId, NamespaceMet
     if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
       builder.setSchedulerQueueName(config.getSchedulerQueueName());
     }
-
+    if (config != null && !Strings.isNullOrEmpty(config.getRootDirectory())) {
```

Comment:
Instead of checking if its non-null and non-empty, I think we should check inequality with the existing value.